addon/modifiers/dropdown-trigger.ts

```diff
@@ -0,0 +1,165 @@
+import Modifier from 'ember-modifier'
+import { action } from '@ember/object'
+import hasMoved from '../utils/has-moved'
+import { Dropdown } from '../components/basic-dropdown'
+
+interface Args {
+  positional: []
+  named: {
+    dropdown: Dropdown
+    eventType: 'click' | 'mousedown'
+    stopPropagation: boolean
+    onBlur?: (dropdown?: Dropdown, event?: FocusEvent) => void
+    onClick?: (dropdown?: Dropdown, event?: MouseEvent) => void
+    onFocus?: (dropdown?: Dropdown, event?: FocusEvent) => void
+    onFocusIn?: (dropdown?: Dropdown, event?: FocusEvent) => void
+    onFocusOut?: (dropdown?: Dropdown, event?: FocusEvent) => void
+    onKeyDown?: (dropdown?: Dropdown, event?: KeyboardEvent) => void
+    onMouseDown?: (dropdown?: Dropdown, event?: MouseEvent) => void
+    onMouseEnter?: (dropdown?: Dropdown, event?: MouseEvent) => void
+    onMouseLeave?: (dropdown?: Dropdown, event?: MouseEvent) => void
+    onTouchEnd?: (dropdown?: Dropdown, event?: TouchEvent) => void
+  }
+}
+
+export default class DropdownTriggerModifier extends Modifier<Args> {
+  private toggleIsBeingHandledByTouchEvents = false
+
+  private touchMoveEvent?: TouchEvent
+
+  didInstall() {
+    this.element.setAttribute('data-ebd-id', `${this.args.named.dropdown.uniqueId}-trigger`)
+
+    this.element.addEventListener('click', this.onClick, true)
+    this.element.addEventListener('mousedown', this.onMouseDown, true)
+    this.element.addEventListener('keydown', this.onKeyDown, true)
+    this.element.addEventListener('touchstart', this.onTouchStart, true)
+    this.element.addEventListener('touchend', this.onTouchEnd, true)
+  }
+
+  @action
+  onMouseDown(e: MouseEvent): void {
+    if (this.args.named.dropdown.disabled) {
+      return
+    }
```

Comment:
I didn't get into this modifier code too thoroughly, I just cut/pasted from the trigger component and gave it a once over.